src/bluesearch/database/article.py

```diff
@@ -189,6 +237,59 @@ def paragraphs(self) -> Generator[tuple[str, str], None, None]:
             caption = " ".join(self._element_to_str(c) for c in caption_elements)
             yield "Table Caption", caption
 
+    @property
+    def pubmed_id(self) -> Optional[str]:
+        """Get Pubmed ID.
+
+        Returns
+        -------
+        str or None
+            Pubmed ID if specified, otherwise None.
+        """
+        return self.ids.get("pmid")
+
+    @property
+    def pmc_id(self) -> Optional[str]:
+        """Get PMC ID.
+
+        Returns
+        -------
+        str or None
+            PMC ID if specified, otherwise None.
+        """
+        return self.ids.get("pmc")
+
+    @property
+    def doi(self) -> Optional[str]:
+        """Get DOI.
+
+        Returns
+        -------
+        str or None
+            DOI if specified, otherwise None.
+        """
+        return self.ids.get("doi")
+
+    def get_ids(self) -> dict[str, str]:
+        """Get all specified IDs of the paper.
+
+        Returns
+        -------
+        ids : dict
+            Dictionary whose keys are ids type and value are ids values.
+        """
+        ids = {}
+        article_ids = self.content.findall("./front/article-meta/article-id")
+
+        for article_id in article_ids:
+
+            if "pub-id-type" not in article_id.attrib.keys():
+                continue
```

Comment:
In which context would this case happen?

Reply:
I am not about a concrete explanation for this, but I had the issue once.